sources/web/datalab/server.ts

```diff
@@ -164,12 +164,13 @@ function handleRequest(request: http.ServerRequest,
       (path.indexOf('/terminals') == 0) ||
       (path.indexOf('/sessions') == 0)) {
 
-    if (path.indexOf('/tree') == 0) {
-      const filePath = '/content' + path.substr('/tree'.length);
+    if (path.indexOf('/api/contents') == 0) {
+      const subPath = path.substr('/api/contents'.length);
+      const filePath = path_.join('/content', subPath);
```

Comment:
Using both path and path_ seems like asking for trouble. How about renaming path to requestPath?